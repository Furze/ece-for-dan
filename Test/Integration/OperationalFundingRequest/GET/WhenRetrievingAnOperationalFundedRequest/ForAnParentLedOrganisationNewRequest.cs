using System;
using System.Collections.Generic;
using MoE.ECE.Domain.Read.Model.OperationalFunding;
using MoE.ECE.Integration.Tests.Chapter;
using MoE.ECE.Integration.Tests.Infrastructure;
using Shouldly;
using Xunit;
using Xunit.Abstractions;

namespace MoE.ECE.Integration.Tests.OperationalFundingRequest.GET.WhenRetrievingAnOperationalFundedRequest
{
    public class ForAnParentLedOrganisationNewRequest : IntegrationTestBase<ECEStoryBook, ECEStoryData>
    {
        public ForAnParentLedOrganisationNewRequest(RunOnceBeforeAllTests testSetUp, ITestOutputHelper output, TestState<ECEStoryBook, ECEStoryData> testState) : base(testSetUp, output, testState)
        {
        }

        private Guid _businessEntityId;
        private int _revisionNumber;

        protected override void Arrange()
        {
            Given
                .An_ParentLed_rs7_application_has_been_received()
                .UseResult(created =>
                {
                    _businessEntityId = created.BusinessEntityId;
                    _revisionNumber = created.RevisionNumber;
                });

            And
                .An_ParentLed_rs7_application_has_been_received(updated =>
            {
                updated.BusinessEntityId = _businessEntityId;
                updated.RevisionNumber = _revisionNumber + 1;
            });
        }

        protected override void Act()
        {
            When.Get($"api/operational-funding-requests?business-entity-id={_businessEntityId}&revision-number={_revisionNumber}");
        }

        [Fact]
        public void ThenTheResponseShouldBeOk()
        {
            Then.Response.ShouldBe.Ok();
        }

        [Fact]
        public void ThenTheResponseShouldContain1Item()
        {
            var result = Then.Response.Content<ICollection<OperationalFundingRequestModel>>();
            
            result.Count.ShouldBe(1);
        }
    }
}