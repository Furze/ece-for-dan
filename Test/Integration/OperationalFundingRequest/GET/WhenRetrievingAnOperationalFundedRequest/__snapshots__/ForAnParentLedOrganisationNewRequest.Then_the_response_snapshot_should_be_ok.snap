[
  {
    "EntitlementMonths": [
      {
        "MonthName": "February 2020",
        "MonthNumber": 2,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 20,
        "AllDayNonCertificatedTeacherHours": 40,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 20,
        "WashUpPlusTen": null,
        "WashUpUnderTwo": null,
        "WashUpTwoAndOver": null,
        "WashUpTwentyHours": 0.0,
        "TotalEntitlement": null,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": null,
        "TotalEntitlementTwoAndOver": null,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": null,
        "TotalWashUp": null,
        "FundingComponents": []
      },
      {
        "MonthName": "March 2020",
        "MonthNumber": 3,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 22,
        "AllDayNonCertificatedTeacherHours": 44,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 22,
        "WashUpPlusTen": null,
        "WashUpUnderTwo": null,
        "WashUpTwoAndOver": null,
        "WashUpTwentyHours": 0.0,
        "TotalEntitlement": null,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": null,
        "TotalEntitlementTwoAndOver": null,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": null,
        "TotalWashUp": null,
        "FundingComponents": []
      },
      {
        "MonthName": "April 2020",
        "MonthNumber": 4,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 22,
        "AllDayNonCertificatedTeacherHours": 44,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 22,
        "WashUpPlusTen": null,
        "WashUpUnderTwo": null,
        "WashUpTwoAndOver": null,
        "WashUpTwentyHours": 0.0,
        "TotalEntitlement": null,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": null,
        "TotalEntitlementTwoAndOver": null,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": null,
        "TotalWashUp": null,
        "FundingComponents": []
      },
      {
        "MonthName": "May 2020",
        "MonthNumber": 5,
        "Year": 2020,
        "AllDayCertificatedTeacherHours": 21,
        "AllDayNonCertificatedTeacherHours": 42,
        "SessionalCertificatedTeacherHours": 0,
        "SessionalNonCertificatedTeacherHours": 0,
        "TotalWorkingDays": 21,
        "WashUpPlusTen": null,
        "WashUpUnderTwo": null,
        "WashUpTwoAndOver": null,
        "WashUpTwentyHours": 0.0,
        "TotalEntitlement": null,
        "TotalFundsAdvanced": 0.0,
        "TotalEntitlementUnderTwo": null,
        "TotalEntitlementTwoAndOver": null,
        "TotalEntitlementTwentyHours": 0.0,
        "TotalEntitlementPlusTen": null,
        "TotalWashUp": null,
        "FundingComponents": []
      }
    ],
    "OrganisationId": 258,
    "BusinessEntityId": "9e7411cd-29cd-4e88-b769-6a969807202b",
    "RequestId": "43001",
    "FundingPeriodMonth": "July",
    "FundingYear": 2021,
    "RevisionNumber": 1,
    "TotalWashUp": 0.0,
    "TotalAdvance": 0.0,
    "AdvanceMonths": [
      {
        "MonthName": "July 2020",
        "MonthNumber": 7,
        "Year": 2020,
        "ParentLedWorkingDays": 0,
        "AllDayWorkingDays": 2,
        "SessionalWorkingDays": null,
        "AmountPayablePlusTen": 0.0,
        "AmountPayableUnderTwo": 0.0,
        "AmountPayableTwoAndOver": 0.0,
        "AmountPayableTwentyHours": 0.0,
        "TotalDays": 2,
        "TotalAdvance": 0.0,
        "FundingComponents": []
      }
    ],
    "MatchingAdvanceMonths": [],
    "Equity": null
  }
]
